jwala-tomcat/src/main/resources/data/scripts/thread-dump.sh

```diff
@@ -8,19 +8,28 @@ esac
 export JAVA_HOME=$1
 export JVM_INSTANCE_DIR=$2
 export JVM_NAME=$3
-
-if [ ! -d ${JVM_INSTANCE_DIR}/thread_dump_reports} ]; then
-      mkdir -p ${JVM_INSTANCE_DIR}/thread_dump_reports
-fi
+export DATA_DIR=$4
+export DUMP_FILE=$5
 
 current_time=$(date +"%m-%d-%Y-%T")
 
 if $linux; then
 	echo $(<${JVM_INSTANCE_DIR}/logs/catalina.pid)
-	/usr/bin/sudo -u tomcat ${JAVA_HOME}/bin/jstack $(<${JVM_INSTANCE_DIR}/logs/catalina.pid) 2>&1 | tee  ${JVM_INSTANCE_DIR}/thread_dump_reports/thread_dump_$current_time
+	thread_dump_output=$(/usr/bin/sudo -u tomcat ${JAVA_HOME}/bin/jstack $(<${JVM_INSTANCE_DIR}/logs/catalina.pid) 2>&1 )
+    if echo "$thread_dump_output" | grep -iqF "Full thread dump"; then
+	   echo "$thread_dump_output" |tee ${DATA_DIR}/$DUMP_FILE
+	   echo "Creating thread-dump file at location: ${DATA_DIR}/$DUMP_FILE, "
+	fi
+
 fi
 
 if $cygwin; then
     export JVMPID=`sc queryex $JVM_NAME | /usr/bin/grep PID | /usr/bin/awk '{ print $3 }'`
-	${JAVA_HOME}/bin/jstack -l ${JVMPID} 2>&1 |tee ${JVM_INSTANCE_DIR}/thread_dump_reports/thread_dump_$current_time
-fi
\ No newline at end of file
+    thread_dump_output=$(${JAVA_HOME}/bin/jstack -l ${JVMPID} 2>&1 )
+    if echo "$thread_dump_output" | grep -iqF "Full thread dump"; then
+       echo "$thread_dump_output" |tee ${DATA_DIR}/$DUMP_FILE
+	   echo "Creating Thread Dump file at location : ${DATA_DIR}/$DUMP_FILE, "
+	fi
```

Comment:
Same thing here. If the success condition isn't met, then by default it failed - but we won't know why because we don't capture any of the unexpected content.